src/main/java/ch/naviqore/raptor/router/RaptorConnection.java

```diff
@@ -69,8 +71,8 @@ public String getToStopId() {
     }
 
     @Override
-    public int getDuration() {
-        return getArrivalTime() - getDepartureTime();
+    public int getDurationInSeconds() {
+        return (int) Duration.between(getDepartureTime(), getArrivalTime()).getSeconds();
```

Comment:
This is critical. int could not be enough. There might scenarios where we have timetables afer January 19 , 2038. Which is actually the case in our simulations at work.

Maybe the data source is not GTFS in this cas, but I think since we are building a new routing service, it should be able to handle future timetables.

Reply:
In this specific case of duration this is not critical unless you want to calculate the duration of a journey that started in 1970 and ended 70 years later. For the date in general I agree, we can convert our unix timestamps to long in this case.

Reply:
Ah sorry, wrong location in the code... im worried about future timetables after 2040...